Flaky-test: PersistentFailoverE2ETest.testSimpleConsumerEventsWithPartition

**Describe the bug**

```
2020-01-07T14:51:53.1621254Z [INFO] Results:
2020-01-07T14:51:53.1621414Z [INFO]
2020-01-07T14:51:53.1624040Z [ERROR] Failures:
2020-01-07T14:51:53.1624466Z [ERROR] org.apache.pulsar.broker.service.PersistentFailoverE2ETest.testActiveConsumerFailoverWithDelay(org.apache.pulsar.broker.service.PersistentFailoverE2ETest)
2020-01-07T14:51:53.1626273Z [INFO]   Run 1: PASS
2020-01-07T14:51:53.1685948Z [ERROR]   Run 2: PersistentFailoverE2ETest.testActiveConsumerFailoverWithDelay:491->MockedPulsarServiceBaseTest.restartBroker:185->MockedPulsarServiceBaseTest.startBroker:195->MockedPulsarServiceBaseTest.startBroker:214 » PulsarServer
2020-01-07T14:51:53.1686338Z [INFO]
2020-01-07T14:51:53.1686659Z [ERROR] org.apache.pulsar.broker.service.PersistentFailoverE2ETest.testSimpleConsumerEventsWithPartition(org.apache.pulsar.broker.service.PersistentFailoverE2ETest)
2020-01-07T14:51:53.1687004Z [INFO]   Run 1: PASS
2020-01-07T14:51:53.1687475Z [ERROR]   Run 2: PersistentFailoverE2ETest.testSimpleConsumerEventsWithPartition » ServerSideError
2020-01-07T14:51:53.1687598Z [INFO]
2020-01-07T14:51:53.1687853Z [ERROR] org.apache.pulsar.broker.service.PersistentFailoverE2ETest.testSimpleConsumerEventsWithoutPartition(org.apache.pulsar.broker.service.PersistentFailoverE2ETest)
2020-01-07T14:51:53.1688003Z [INFO]   Run 1: PASS
2020-01-07T14:51:53.1688400Z [ERROR]   Run 2: PersistentFailoverE2ETest.testSimpleConsumerEventsWithoutPartition:159 » PulsarClient
2020-01-07T14:51:53.1688638Z [INFO]
2020-01-07T14:51:53.1688797Z [INFO]
2020-01-07T14:51:53.1688922Z [ERROR] Tests run: 1408, Failures: 3, Errors: 0, Skipped: 0
2020-01-07T14:51:53.1689026Z [INFO]
2020-01-07T14:51:53.1703494Z [INFO] ------------------------------------------------------------------------
2020-01-07T14:51:53.1703742Z [INFO] BUILD FAILURE
2020-01-07T14:51:53.1704113Z [INFO] ------------------------------------------------------------------------
2020-01-07T14:51:53.1707623Z [INFO] Total time:  18:18 min
2020-01-07T14:51:53.1718769Z [INFO] Finished at: 2020-01-07T14:51:53Z
2020-01-07T14:51:53.1719163Z [INFO] ------------------------------------------------------------------------
2020-01-07T14:51:53.1734935Z [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M3:test (default-test) on project pulsar-broker: There are test failures.
2020-01-07T14:51:53.1735182Z [ERROR]
2020-01-07T14:51:53.1736569Z [ERROR] Please refer to /home/runner/work/pulsar/pulsar/pulsar-broker/target/surefire-reports for the individual test results.
2020-01-07T14:51:53.1737042Z [ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
2020-01-07T14:51:53.1737361Z [ERROR] -> [Help 1]
2020-01-07T14:51:53.1737450Z [ERROR]
2020-01-07T14:51:53.1737777Z [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
2020-01-07T14:51:53.1738095Z [ERROR] Re-run Maven using the -X switch to enable full debug logging.
2020-01-07T14:51:53.1803454Z [ERROR]
2020-01-07T14:51:53.1803674Z [ERROR] For more information about the errors and possible solutions, please read the following articles:
2020-01-07T14:51:53.1803874Z [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
2020-01-07T14:51:53.1921294Z ##[error]Process completed with exit code 1.
```

**To Reproduce**
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Desktop (please complete the following information):**
 - OS: [e.g. iOS]

**Additional context**
Add any other context about the problem here.

[Recent failure](https://github.com/apache/pulsar/pull/11046/checks?check_run_id=2894997728#step:10:1084):
```
Error:  Tests run: 6, Failures: 1, Errors: 0, Skipped: 3, Time elapsed: 55.531 s <<< FAILURE! - in org.apache.pulsar.broker.service.PersistentFailoverE2ETest
Error:  testSimpleConsumerEventsWithoutPartition(org.apache.pulsar.broker.service.PersistentFailoverE2ETest)  Time elapsed: 0.638 s  <<< FAILURE!
java.lang.AssertionError: expected [0] but found [20]
	at org.testng.Assert.fail(Assert.java:99)
	at org.testng.Assert.failNotEquals(Assert.java:1037)
	at org.testng.Assert.assertEqualsImpl(Assert.java:140)
	at org.testng.Assert.assertEquals(Assert.java:122)
	at org.testng.Assert.assertEquals(Assert.java:797)
	at org.testng.Assert.assertEquals(Assert.java:807)
	at org.apache.pulsar.broker.service.PersistentFailoverE2ETest.testSimpleConsumerEventsWithoutPartition(PersistentFailoverE2ETest.java:204)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PersistentFailoverE2ETest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PersistentFailoverE2ETest.java
@@ -52,6 +52,7 @@ import org.apache.pulsar.client.impl.TopicMessageImpl;
 import org.apache.pulsar.common.api.proto.CommandSubscribe.SubType;
 import org.apache.pulsar.common.naming.TopicName;
 import org.apache.pulsar.common.util.FutureUtil;
+import org.awaitility.Awaitility;
 import org.testng.Assert;
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.BeforeClass;
@@ -184,8 +185,9 @@ public class PersistentFailoverE2ETest extends BrokerTestBase {
 
         rolloverPerIntervalStats();
 
-        assertEquals(subRef.getNumberOfEntriesInBacklog(false), numMsgs);
-        Thread.sleep(ASYNC_EVENT_COMPLETION_WAIT);
+        Awaitility.await().untilAsserted(() -> {
+            assertEquals(subRef.getNumberOfEntriesInBacklog(false), numMsgs);
+        });
 
         // 3. consumer1 should have all the messages while consumer2 should have no messages
         Message<byte[]> msg = null;
@@ -200,8 +202,9 @@ public class PersistentFailoverE2ETest extends BrokerTestBase {
         rolloverPerIntervalStats();
 
         // 4. messages deleted on individual acks
-        Thread.sleep(ASYNC_EVENT_COMPLETION_WAIT);
-        assertEquals(subRef.getNumberOfEntriesInBacklog(false), 0);
+        Awaitility.await().untilAsserted(() -> {
+            assertEquals(subRef.getNumberOfEntriesInBacklog(false), 0);
+        });
 
         for (int i = 0; i < numMsgs; i++) {
             String message = "my-message-" + i;
@@ -224,10 +227,12 @@ public class PersistentFailoverE2ETest extends BrokerTestBase {
             // do not ack
         }
         consumer1.close();
-        Thread.sleep(CONSUMER_ADD_OR_REMOVE_WAIT_TIME);
 
-        verifyConsumerActive(listener2, -1);
-        verifyConsumerNotReceiveAnyStateChanges(listener1);
+        Awaitility.await().untilAsserted(() -> {
+            verifyConsumerActive(listener2, -1);
+            verifyConsumerNotReceiveAnyStateChanges(listener1);
+        });
+
         for (int i = 5; i < numMsgs; i++) {
             msg = consumer2.receive(1, TimeUnit.SECONDS);
             Assert.assertNotNull(msg);
@@ -237,8 +242,10 @@ public class PersistentFailoverE2ETest extends BrokerTestBase {
         Assert.assertNull(consumer2.receive(100, TimeUnit.MILLISECONDS));
 
         rolloverPerIntervalStats();
-        Thread.sleep(ASYNC_EVENT_COMPLETION_WAIT);
-        assertEquals(subRef.getNumberOfEntriesInBacklog(false), 0);
+        Awaitility.await().untilAsserted(() -> {
+            assertEquals(subRef.getNumberOfEntriesInBacklog(false), 0);
+
+        });
 
         // 8. unsubscribe not allowed if multiple consumers connected
         try {
@@ -257,9 +264,9 @@ public class PersistentFailoverE2ETest extends BrokerTestBase {
             fail("Should not fail", e);
         }
 
-        Thread.sleep(ASYNC_EVENT_COMPLETION_WAIT);
-        subRef = topicRef.getSubscription(subName);
-        assertNull(subRef);
+        Awaitility.await().untilAsserted(() -> {
+            assertNull(topicRef.getSubscription(subName));
+        });
 
         producer.close();
         consumer2.close();

